crates/bevy_render/src/render_asset.rs

```diff
@@ -39,23 +39,63 @@ pub trait RenderAsset: Asset {
     ) -> Result<Self::PreparedAsset, PrepareAssetError<Self::ExtractedAsset>>;
 }
 
+#[derive(Clone, Hash, Debug, PartialEq, Eq, SystemLabel)]
+pub enum PrepareAssetLabel {
+    PreAssetPrepare,
+    AssetPrepare,
+    PostAssetPrepare,
+}
+
+impl Default for PrepareAssetLabel {
+    fn default() -> Self {
+        Self::AssetPrepare
+    }
+}
+
 /// This plugin extracts the changed assets from the "app world" into the "render world"
 /// and prepares them for the GPU. They can then be accessed from the [`RenderAssets`] resource.
 ///
 /// Therefore it sets up the [`RenderStage::Extract`](crate::RenderStage::Extract) and
 /// [`RenderStage::Prepare`](crate::RenderStage::Prepare) steps for the specified [`RenderAsset`].
-pub struct RenderAssetPlugin<A: RenderAsset>(PhantomData<fn() -> A>);
+pub struct RenderAssetPlugin<A: RenderAsset> {
+    prepare_asset_label: PrepareAssetLabel,
+    phantom: PhantomData<fn() -> A>,
+}
+
+impl<A: RenderAsset> RenderAssetPlugin<A> {
+    pub fn with_prepare_asset_label(prepare_asset_label: PrepareAssetLabel) -> Self {
```

Comment:
Any better suggestions for function name?

Reply:
Seems good to me!